solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedFeatureStore.java

```diff
@@ -80,48 +83,32 @@ protected void onManagedDataLoadedFromStorage(NamedList<?> managedInitArgs,
   }
 
   public void update(Map<String,Object> map) {
-    final String name = (String) map.get(CommonLTRParams.MODEL_NAME);
-    final String type = (String) map.get(CommonLTRParams.MODEL_CLASS);
-    final String store = (String) map.get(CommonLTRParams.MODEL_FEATURE_STORE);
-
-    @SuppressWarnings("unchecked")
-    final Map<String,Object> paramsMap = (Map<String,Object>) map.get(CommonLTRParams.MODEL_PARAMS);
-
     try {
-
-      addFeature(name, type, store, paramsMap);
+      addFeature(map);
     } catch (final FeatureException e) {
       throw new SolrException(ErrorCode.BAD_REQUEST, e);
     }
   }
 
-  public synchronized void addFeature(String name, String type,
-      String featureStore, Map<String,Object> params)
+  public void addFeature(Map<String,Object> map)
```

Comment:
No, now inlined the only call to the function and removed the function, good catch.